enhancements/node-tuning/pao-in-nto.md

```diff
@@ -0,0 +1,317 @@
+
+### Implementation Details/Notes/Constraints
+
+- Existing api compatibility practices:
+PAO API has been treated as Tier 1 for multiple releases already (since 4.4).
+
+- Enumerate list of performance profile owned objects:
+MachineConfig, KubeletConfiguration, Tuned (see [diagram](https://github.com/openshift-kni/performance-addon-operators/blob/master/docs/interactions/diagram.png)).
+All activity happening on the cluster or nodes is indirect via other components (MCO, NTO) reacting to changes in objects owned by PAO.
+
+- Enumerate list of webhooks:
+Validation and conversion webhook (TBD)
+
+- Updates to openshift conformance test suite if included by default:
+All basic checks are doable on all linux platforms except when real-time kernel is needed. That case used to be only supported on GCP (and bare metal).
+Measuring real latency requires bare metal, but is not needed for conformance.
+
+
+### Risks and Mitigations
+
+What are the risks of this proposal and how do we mitigate. Think broadly. For
+example, consider both security and how this will impact the larger OKD
+ecosystem.
+
+How will security be reviewed and by whom? How will UX be reviewed and by whom?
+
+Consider including folks that also work outside your immediate sub-project.
+
+## Design Details
+
+### Open Questions
+
+- CI: Integration with NTO is currently TBD.
+
+- CI: One of the test suites (4_latency) implements an opinionated latency testing suite to get comparable results from customers and to prevent configuration / command line invocation mistakes. Keeping it in PAO/NTO or moving it into cnf-tests is TBD.
+
+- Maintainability: There are two options:
+  - Maintaining a separate copy of the code in the NTO repository.
+  - Vendoring the current PAO code as a "library" until we can stop supporting the standalone mode (4.6+).
+  From OCP perspective it makes little difference as it would just be part of NTO release process indeed.
+
+- API definitions: They would need to go into openshift/api but at this moment NTO does include its own API definitions under openshift/api.
+The NTO API definitions transition into openshift/api should be considered to be done at the time of PAO/NTO merge.
+
+- List of pertinent topology options (control plane and infrastructure) where this operator is pertinent:
+The assumption is that PAO is not installed at all when control plane topology is "external" and the infrastructure topology is not pertinent to the function
+since resources leveraged by the PAO are all managed externally when in this mode (e.g. node configuration, kubelet configuration).
+PAO is a control plane component that modifies the worker nodes (unless in single node mode where it affects masters).
+
+### Upgrade
+
+- PAO has been installed via OLM since 4.4. Once it moves to core we will have to figure out the upgrade procedure, because OLM will not remove the optional operator by itself.
+
+  - Implementations options:
```

Comment:
We should pick one of these to describe here and document the others in the alternatives section below.